File: frontend/src/types/balance.ts
export interface BalanceResponse {
  salary: string;
  total_expenses_this_month: string;
  remaining_balance: string;
}

export interface CategoryTotal {
  category: string;
  total: string;
}

export interface MonthlySummary {
  year: number;
  month: number;
  salary: string;
  total_expenses: string;
  remaining_balance: string;
  expense_count: number;
  by_category: CategoryTotal[];
}

File: backend/app/schemas/balance.py
from decimal import Decimal

from pydantic import BaseModel


class CategoryTotal(BaseModel):
    category: str
    total: Decimal


class MonthlySummary(BaseModel):
    year: int
    month: int
    salary: Decimal
    total_expenses: Decimal
    remaining_balance: Decimal
    expense_count: int
    by_category: list[CategoryTotal]


class BalanceResponse(BaseModel):
    salary: Decimal
    total_expenses_this_month: Decimal
    remaining_balance: Decimal
